crates/red_knot_project/src/metadata.rs

```diff
@@ -62,22 +65,36 @@ impl ProjectMetadata {
 
     /// Loads a project from a set of options with an optional pyproject-project table.
     pub(crate) fn from_options(
-        options: Options,
+        mut options: Options,
         root: SystemPathBuf,
         project: Option<&Project>,
-    ) -> Self {
+    ) -> Result<Self, TooLargeRequiresPythonError> {
         let name = project
-            .and_then(|project| project.name.as_ref())
-            .map(|name| Name::new(&***name))
+            .and_then(|project| project.name.as_deref())
+            .map(|name| Name::new(&**name))
             .unwrap_or_else(|| Name::new(root.file_name().unwrap_or("root")));
 
-        // TODO(https://github.com/astral-sh/ruff/issues/15491): Respect requires-python
-        Self {
+        // If the project doesn't specify a python version but the `project.requires-python` field is set, resolve the version
```

Comment:
Maybe something like
```suggestion
        // If the options don't specify a python version but the `project.requires-python` field is set and includes a lower bound, use that instead
```